app/src/main/java/com/nmakademija/nmaakademija/fragment/UsersFragment.java

```diff
@@ -40,11 +41,38 @@
 
     private Spinner spinner;
     private AppEvent appEvent;
+    private ArrayList<User> users;
+    private ArrayList<Section> sections;
+    private int lastFilter = 0;
+
+    public String EXTRA_USERS = "users";
+    public String EXTRA_SECTIONS = "sections";
+    public String EXTRA_LAST_FILTER = "pager_filter";
 
     public static UsersFragment getInstance() {
         return new UsersFragment();
     }
 
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if(savedInstanceState != null){
+            users = savedInstanceState.getParcelableArrayList(EXTRA_USERS);
+            sections = savedInstanceState.getParcelableArrayList(EXTRA_SECTIONS);
+            lastFilter = savedInstanceState.getInt(EXTRA_LAST_FILTER);
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        outState.putParcelableArrayList(EXTRA_USERS, users);
+        outState.putParcelableArrayList(EXTRA_SECTIONS, sections);
+        outState.putInt(EXTRA_LAST_FILTER, spinner.getSelectedItemPosition());
```

Comment:
spinner might be null here. For example if request is not finished. Move spinner variable initialisation out of `setData`